### ISSUE TITLE
min and max version not shown in cloud/apps/%app% XML

### ISSUE THREAD

**Issue by phil-davis** [owncloud/core#29229]
### Steps to reproduce
1. Browse to examples like these (put your oC server in place of localhost:8080)
```
http://localhost:8080/ocs/v1.php/cloud/apps/comments
http://localhost:8080/ocs/v1.php/cloud/apps/files
http://localhost:8080/ocs/v2.php/cloud/apps/comments
http://localhost:8080/ocs/v2.php/cloud/apps/files
```

The returned XML has a dependencies section. This (along with the other sections) comes from the ``info.xml`` for the app.

### Expected behaviour
I was thinking that min-version and max-version would be displayed somewhere (since they are in the info.xml).
```
<dependencies>
  <owncloud min-version="10.0" max-version="10.0"/>
</dependencies>
```

### Actual behaviour
min-version and max-version are not seen:
```
<dependencies>
  <owncloud>
  </owncloud>
</dependencies>
```

### Server configuration
**Operating system**:
Ubuntu 16.04
**Web server:**
PHP dev server
**Database:**
any
**PHP version:**
7.0.22
**ownCloud version:** (see ownCloud admin page)
10.0.3
**Updated from an older ownCloud or fresh install:**
git clone
**Where did you install ownCloud from:**
git clone

### Logs
This was noticed because of the output from an integration test case:
```
  Scenario: get app info                      # /var/lib/jenkins/workspace/owncloud-core_core_PR-29211-EWXGKJMBFX2VD4V6UIXOO2YK4FEFEBD4KRQFHV2OZBXIXWFSS3YQ/tests/integration/features/provisioning-v1.feature:299
    Given as an "admin"     
                  # FeatureContext::asAn()
[Thu Oct 12 03:01:18 2017] XMLWriter::writeAttribute() expects parameter 2 to be string, array given at /var/lib/jenkins/workspace/owncloud-core_core_PR-29211-EWXGKJMBFX2VD4V6UIXOO2YK4FEFEBD4KRQFHV2OZBXIXWFSS3YQ/lib/private/legacy/api.php#439
[Thu Oct 12 03:01:18 2017] ::1:42791 [200]: /ocs/v1.php/cloud/apps/files
    When sending "GET" to "/cloud/apps/files" # FeatureContext::sendingTo()
    Then the OCS status code should be "100"  # FeatureContext::theOCSStatusCodeShouldBe()
    And the HTTP status code should be "200"  # FeatureContext::theHTTPStatusCodeShouldBe()
```
A warning is logged about ``XMLWriter::writeAttribute() expects parameter 2 to be string, array given``

**Comment by phil-davis** [comment]
A typical ``info.xml`` has a section like:
```
	<dependencies>
		<owncloud min-version="10.0" max-version="10.0" />
	</dependencies>
```
``min-version`` and ``max-version`` get seen as attributes of ``owncloud`` (I guess they are). The internal code that processes to a PHP object and back to XML has a problem with that.

**Comment by phil-davis** [comment]
Fixed by #29230
